code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/model/SubmissionsGraph.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.code.submissions.clustering.model
+
+import org.jetbrains.research.code.submissions.clustering.load.AbstractUnifier
+import org.jgrapht.Graph
+import org.jgrapht.graph.DefaultWeightedEdge
+import org.jgrapht.graph.SimpleDirectedWeightedGraph
+
+data class SubmissionsGraph(private val graph: Graph<SubmissionsNode, DefaultWeightedEdge>) {
+    fun buildStringRepresentation() = graph.toString()
+}
+
+class GraphBuilder(private val unifier: AbstractUnifier) {
+    private val graph: Graph<SubmissionsNode, DefaultWeightedEdge> =
+        SimpleDirectedWeightedGraph(DefaultWeightedEdge::class.java)
+
+    fun add(submission: Submission) {
+        unifier.run {
+            val unifiedSubmission = submission.unify()
+            // Check if suitable vertex already exists
+            val vertex = graph.vertexSet().find { it.code == unifiedSubmission.code }
+            vertex?.let {
+                // Update vertex
+                graph.removeVertex(vertex)
+                graph.addVertex(SubmissionsNode(vertex, unifiedSubmission))
```

Comment:
It looks a little bit strange, can we replace the vertex or update the existing one?

Reply:
Changed function by updating existing vertex instead of removing old vertex and adding new